### PR TITLE
Add IndexOptions to IndexInfo json response

### PR DESCRIPTION
**Pull request by codysoyland**
This adds IndexOptions to responses that serialize IndexInfo, including /schema, /index, and /index/{index}.

This came up during discussion on #1540.

### CODE REVIEW

File: index.go [hunk 1]
```diff
@@ -95,7 +95,7 @@ func (i *Index) Options() IndexOptions {
 }
 
 func (i *Index) options() IndexOptions {
-	return IndexOptions{}
+	return IndexOptions{i.keys}
```
Comment (jaffee) [inline index.go]: should be `IndexOptions{Keys: i.keys}` no?

File: http/handler_internal_test.go [hunk 2]
```diff
@@ -31,6 +31,7 @@ func TestPostIndexRequestUnmarshalJSON(t *testing.T) {
 		err      string
 	}{
 		{json: `{"options": {}}`, expected: postIndexRequest{Options: pilosa.IndexOptions{}}},
+		{json: `{"options": {"keys": true}}`, expected: postIndexRequest{Options: pilosa.IndexOptions{true}}},
```
Comment (jaffee) [inline http/handler_internal_test.go]: also here, should use `Keys:`

File: index.go (hunk 1 above)
Comment (codysoyland) [reply index.go]: Sure, no big reason not to use the field labels. I think the linter only complains when they are cross-package, but I agree they are good to use. Pushed another commit that adds them.

Comment (jaffee) [reply index.go]: ah, I thought it complained all the time - anywho

**Pull request merged**